test/test_dht.cpp

```diff
@@ -3697,10 +3698,13 @@ TORRENT_TEST(sample_infohashes)
 	udp::endpoint const ep2 = rand_udp_ep(rand_v4);
 
 	t.dht_node.sample_infohashes(initial_node, items[0].target,
-		[h1, ep1, h2, ep2](time_duration interval, int num
+		[nid, h1, ep1, h2, ep2](node_id const& node_id
+			, time_duration interval
+			, int num
 			, std::vector<sha1_hash> samples
 			, std::vector<std::pair<sha1_hash, udp::endpoint>> const& nodes)
 	{
+		TEST_EQUAL(node_id, nid);
```

Comment:
I think you want to compare `node_id` against the node ID that was set up as part of the `dht_test_setup`. I think you can get to this as: `t.dht_node.nid()`

Reply:
oh, I didn't look at this closely enough... this is what you're doing already..

Reply:
ok line 3741, that's where we send the response to this message. after `msg_args()` add a line `.nid(id)`, (with whatever `id` you'd want)

Reply:
Ok, now I'm using this to generate a random id
```
node_id nid = generate_random_id();
```
keeping this as is
```
	t.dht_node.sample_infohashes(initial_node, items[0].target,
		[nid, h1, ep1, h2, ep2](node_id const& node_id
			, time_duration interval
			, int num
			, std::vector<sha1_hash> samples
			, std::vector<std::pair<sha1_hash, udp::endpoint>> const& nodes)
	{
		TEST_EQUAL(node_id, nid);
		TEST_EQUAL(total_seconds(interval), 10);
```
and changing this to include the randomly generated nid
```
	send_dht_response(t.dht_node, response, initial_node
		, msg_args()
			.nid(nid)
			.interval(seconds(10))
```

Reply:
It did build fine and checks were passed before rebasing, so it should be fixed now.